src/test/kotlin/io/hsar/wh40k/combatsimulator/logic/TacticalActionStrategyTest.kt

```diff
@@ -0,0 +1,49 @@
+package io.hsar.wh40k.combatsimulator.logic
+
+import io.hsar.wh40k.combatsimulator.model.MapPosition
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.jupiter.api.Test
+
+internal class TacticalActionStrategyTest {
+    private val meleeAttack = ActionOption.MeleeAttack("1d10+4")
+    private val singleRangedAttack = ActionOption.SingleRangedAttack("1d10+2", 30)
+    private val chargeAttack = ActionOption.ChargeAttack("1d10+5")
+    private val aimAction = ActionOption.HalfAim
+
+    @Test
+    fun decideTurnActions() {
+        val world = TestUtils.getGenericTwoUnitWorld(MapPosition(2,3), MapPosition(10,5))
+        val unit = world.friendlyForces[0]
+        val feasibleActions: List<TurnAction> = TacticalActionStrategy.decideTurnActions(world, unit, listOf(meleeAttack, singleRangedAttack, chargeAttack, aimAction))
+        assertThat(feasibleActions.size, equalTo(2))
+        assertThat(feasibleActions[0] is AimAction, equalTo(true))
+        assertThat(feasibleActions[1].action is ActionOption.ChargeAttack, equalTo(true))
```

Comment:
I think that it would be a stronger test if we instantiate an expected `ChargeAttack` instance and assert `equalTo`, and the same for the AimAction above.

Reply:
Tests updated to check the TurnAction actions against the specific object references used.